shared/cluster.go

```diff
@@ -34,52 +35,64 @@ type Pod struct {
 	Node      string
 }
 
-// ManageWorkload creates or deletes a workload based on the action: create or delete.
-func ManageWorkload(action, workload, arch string) (string, error) {
+// ManageWorkload applies or deletes a workload based on the action: apply or delete.
+func ManageWorkload(action string, arch string, workloads ...string) (string, error) {
+	if action != "apply" && action != "delete" {
+		return "", fmt.Errorf("invalid action: %s. Must be 'apply' or 'delete'", action)
+	}
+	
 	var res string
 	var err error
 
-	if action != "create" && action != "delete" {
-		return "", fmt.Errorf("invalid action: %s. Must be 'create' or 'delete'", action)
-	}
-
-	resourceDir := BasePath() + "/distros-test-framework/workloads/amd64/"
-	if arch == "arm64" {
-		resourceDir = BasePath() + "/distros-test-framework/workloads/arm/"
+	resourceDir := BasePath() + "/distros-test-framework/workloads/"
+	if arch != "" {
+		resourceDir += arch
 	}
-
+	
 	files, err := os.ReadDir(resourceDir)
-	if err != nil {
-		err = fmt.Errorf("%s : Unable to read resource manifest file for %s", err, workload)
-		return "", err
-	}
-
-	for _, f := range files {
-		filename := filepath.Join(resourceDir, f.Name())
-		if strings.TrimSpace(f.Name()) == workload {
-			if action == "create" {
-				res, err = createWorkload(workload, filename)
-				if err != nil {
-					return "", fmt.Errorf("failed to create workload %s: %s", workload, err)
-				}
-			} else {
-				err = deleteWorkload(workload, filename)
-				if err != nil {
-					return "", fmt.Errorf("failed to delete workload %s: %s", workload, err)
-				}
+	for _, workload := range workloads {
+		if !fileExists(files, workload) {
+			return "", fmt.Errorf("workload %s not found in %s", workload, resourceDir)
+		}
+		filename := filepath.Join(resourceDir, workload)
+		if action == "apply" {
+			err = applyWorkload(workload, filename)
+			if err != nil {
+				return "", fmt.Errorf("failed to apply workload %s: %s", workload, err)
 			}
-			return res, err
+		} else if action == "delete" {
+			err = deleteWorkload(workload, filename)
+			if err != nil {
+				return "", fmt.Errorf("failed to delete workload %s: %s", workload, err)
+			}
+		} else {
+			return "", fmt.Errorf("workload %s not found in %s", workload, resourceDir)
 		}
 	}
 
-	return "", fmt.Errorf("workload %s not found", workload)
+	return res, err
 }
 
-func createWorkload(workload, filename string) (string, error) {
-	fmt.Println("\nDeploying", workload)
+// applyWorkload applies a workload to the cluster.
+func applyWorkload(workload, filename string) (error) {
+	fmt.Println("\nApplying ", workload)
 	cmd := "kubectl apply -f " + filename + " --kubeconfig=" + KubeConfigFile
+	out, err := RunCommandHost(cmd)
```

Comment:
this RunCommandHost functions, accepts variadcs inputs args. 
you can pass both commands in the same call , if you dont need any extra stuff between both

Reply:
ok will check that

Reply:
Actually I wanted to get some suggestion on this. I want to wait for sometime after the workload is applied as in some cases it takes a while for the container to be created... so I was thinking about using Eventually... wdyt?